api/utils/keys/yubikey.go

```diff
@@ -53,27 +54,13 @@ const (
 // getOrGenerateYubiKeyPrivateKey connects to a connected yubiKey and gets a private key
 // matching the given touch requirement. This private key will either be newly generated
 // or previously generated by a Teleport client and reused.
-func getOrGenerateYubiKeyPrivateKey(touchRequired bool) (*PrivateKey, error) {
-	// TODO(Joerger): Pass ctx through method.
-	ctx := context.TODO()
-
+func getOrGenerateYubiKeyPrivateKey(ctx context.Context, requiredKeyPolicy PrivateKeyPolicy, slot string) (*PrivateKey, error) {
```

Comment:
nit: Why the slot is passed here as string? Can we create a new type `type PIVSlot string` to indicate what that string is?